flink-connector-bigquery-common/src/test/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransformTest.java

```diff
@@ -0,0 +1,633 @@
+                        "value", org.apache.avro.Schema.create(Type.LONG), "", null));
+
+        org.apache.avro.Schema nullableLongListSchema =
+                org.apache.avro.Schema.createUnion(
+                        org.apache.avro.Schema.create(Type.NULL), longListSchema);
+        fields.add(new org.apache.avro.Schema.Field("next", nullableLongListSchema, "", null));
+        longListSchema.setFields(fields);
+
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> AvroToBigQuerySchemaTransform.getBigQuerySchema(longListSchema));
+    }
+
+    /**
+     * Tests Avro record schema with more than 15 levels of nesting. This should throw an Exception.
+     */
+    @Test
+    public void testNestedRecordExceedsLimitThrowsException() {
+        org.apache.avro.Schema nestedSchema =
+                org.apache.avro.Schema.createRecord("NestedRecord", "", "", false);
+        List<org.apache.avro.Schema.Field> fields = new ArrayList<>();
+        org.apache.avro.Schema currentSchema = nestedSchema;
+        for (int i = 0; i < 16; i++) {
+            org.apache.avro.Schema nextSchema =
+                    org.apache.avro.Schema.createRecord("NestedRecord" + i, "", "", false);
+            fields.add(new org.apache.avro.Schema.Field("nestedField", nextSchema, "", null));
+            currentSchema.setFields(fields);
+            currentSchema = nextSchema;
+            fields = new ArrayList<>();
+        }
+
+        org.apache.avro.Schema nestedRecord =
+                org.apache.avro.Schema.createRecord("NestedRecord", "", "", false);
+        ArrayList<org.apache.avro.Schema.Field> nestedFields = new ArrayList<>();
+        nestedFields.add(new org.apache.avro.Schema.Field("nestedField", nestedRecord, "", null));
+        nestedRecord.setFields(nestedFields);
+
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> AvroToBigQuerySchemaTransform.getBigQuerySchema(nestedRecord));
+    }
+
+    /**
+     * Tests Avro schema: { "type": "record", "name": "ArrayRecord", "fields": [ { "name":
+     * "arrayField", "type": { "type": "array", "items": { "type": "array", "items": "int" } } } ]
+     * }. It should throw an exception since recursive Arrays are not supported by BigQuery
+     */
+    @Test
+    public void testArrayOfArraysThrowsException() {
+        org.apache.avro.Schema arrayOfArraysSchema =
+                org.apache.avro.Schema.createArray(
+                        org.apache.avro.Schema.createArray(
+                                org.apache.avro.Schema.create(org.apache.avro.Schema.Type.INT)));
+
+        org.apache.avro.Schema arrayRecord =
+                org.apache.avro.Schema.createRecord("ArrayRecord", "", "", false);
+        ArrayList<org.apache.avro.Schema.Field> arrayFields = new ArrayList<>();
+        arrayFields.add(
+                new org.apache.avro.Schema.Field("arrayField", arrayOfArraysSchema, "", null));
```

Comment:
Would it be possible to use the schema's at TestBigQuerySchema here? 
ik they are in different modules but could we get that here?

Reply:
Importing methods from /test folder will require some efforts. Currently continuing without importing them.